src/aead/aead.rs

```diff
@@ -309,6 +310,19 @@ impl Algorithm {
     pub fn nonce_len(&self) -> usize { NONCE_LEN }
 }
 
+#[derive(Eq, PartialEq)]
+#[allow(non_camel_case_types)]
+enum AlgorithmID {
+    AES_128_GCM,
+    AES_256_GCM,
+    CHACHA20_POLY1305,
+}
+
+impl PartialEq for Algorithm {
+    fn eq(&self, other: &Self) -> bool { self.id == other.id }
+}
+
+impl Eq for Algorithm {}
```

Comment:
I looked into it. It used to work, but now it doesn't, and that is intentional: https://github.com/rust-lang/rust/issues/36830.